File: CivitasRuby/lib/casilla_impuesto.rb
require_relative 'casilla'
module Civitas
  class CasillaImpuesto < Casilla
    def initialize(nombre,cantidad)
      super(nombre)
      @cantidad = cantidad
    end

    def recibe_jugador(iactual,todos)
      if jugador_correcto(iactual,todos)
          informe(iactual,todos)
          todos[iactual].paga_impuesto(@importe)
      end
    end

      def to_s
        "Casilla { \n Nombre: #{@nombre} \n Valor: #{@cantidad} \n}"
      end

  end
end

File: CivitasJava/src/Civitas/CasillaImpuesto.java
package Civitas;

import java.util.ArrayList;

public class CasillaImpuesto extends Casilla {
    private float importe;
    
    CasillaImpuesto(String nombre, float cantidad){
        super(nombre);
        this.importe = cantidad;
        super.tipo = 2;
    }
    
    @Override
    void recibeJugador(int iactual, ArrayList<Jugador> todos) {
        if(super.jugadorCorrecto(iactual,todos)){
            super.informe(iactual,todos);
            todos.get(iactual).pagaImpuesto(importe);                    
        }        
    }
    
    @Override
    public String toString(){
        return "Casilla {" + "\n Nombre: " + super.getNombre() +" Valor " + this.importe +"}\n";
    }
}
